discovery/syncer.go

```diff
@@ -183,11 +184,16 @@ type gossipSyncerCfg struct {
 	// encoding type that we can fit into a single message safely.
 	chunkSize int32
 
-	// sendToPeer is a function closure that should send the set of
-	// targeted messages to the peer we've been assigned to sync the graph
-	// state from.
+	// sendToPeer sends a variadic number of messages to the remote peer.
```

Comment:
initially i thought it might be nice to keep the async version for use in our own `channelGraphSyncer`, since it would be less blocking and more available to receive commands from the the `SyncManager`. it also seemed a more minimal change, though i think it's worth exploring making them all synchronous, especially since we already handle the errors on most of the call sites